src/test/clojure/xtdb/operator/scan_test.clj

```diff
@@ -72,6 +72,21 @@
                      [:scan {:table xt_docs} [foo]]]
                    {:node node}))))))
 
+(t/deftest test-row-copying-different-struct-types-bewteen-chunk-boundaries-3338
+  (with-open [node (xtn/start-node {:indexer {:rows-per-chunk 20}})]
+    (xt/submit-tx node (for [i (range 20)]
+                         [:put-docs :xt_docs {:xt/id i :foo {:bar 42}}]))
+
+    (xt/submit-tx node (for [i (range 20 40)]
+                         [:put-docs :xt_docs {:xt/id i :foo {:bar "forty-two"}}]))
+
+    (t/is (= #{{:foo {:bar 42}} {:foo {:bar "forty-two"}}}
+             (set (tu/query-ra
+                   '[:apply :cross-join {}
```

Comment:
I'm missing the nuance of this being apply cross-join rather than a straight scan?

Reply:
A `scan` doesn't copy. The cross join is using the `MVR` row copier. If you have a simpler example let me know. I can also add a comment if you think that would be helpful.